src/models/state/archival_state.rs

```diff
@@ -718,13 +720,12 @@ impl ArchivalState {
     /// max search depth, as it loads all the blocks in the search path into
     /// memory. A max search depth of 0 means that only the tip is checked.
     pub(crate) async fn get_mutator_set_update_to_tip(
```

Comment:
1. perhaps consider changing the name of this fn.   It seems odd to have a fn prefixed with `get_` that mutates self.

2. just a note regarding the change to &mut self.  any time we perform a lot of db read operations inside a write-lock, it is potentially bad for concurrency as all readers must wait including rpc calls.  Hopefully we can eventually refactor such methods into separate read and write methods where the read method only acquires read lock for gathering updates and the write-lock is held as briefly as possible to write out the updates.  The tricky part though is atomicity over the entire read+write.  (not proposing a change here at this time)

Reply:
It takes a `&mut self` but does not actually mutate it -- as in it's the same mutator set before and after the call. It does persist any cached values but from a mathematical perspective, it's the same mutator set.